ansible/roles/cluster/templates/kubeadm-init.yaml.j2

```diff
@@ -0,0 +1,112 @@
+#jinja2:lstrip_blocks: True
+---
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: InitConfiguration
+bootstrapTokens:
+- token: {{ kubeadm_join_token }}
+  ttl: 1h
+  groups:
+  - system:bootstrappers:kubeadm:default-node-token
+  usages:
+  - signing
+  - authentication
+localAPIEndpoint:
+  advertiseAddress: {{ ansible_default_ipv4.address|default(ansible_all_ipv4_addresses[0]) }}
+  bindPort: {{ cluster_apiserver_bind_port }}
+certificateKey: {{ kubeadm_certificate_key }}
+nodeRegistration:
+  {% if inventory_hostname in groups['masters'] and inventory_hostname not in groups['workers'] %}
```

Comment:
How do you feel about adding labels on worker nodes with `node-role.kubernetes.io/worker` ?